src/main/kotlin/io/github/rybalkinsd/kohttp/dsl/async/HttpPutAsyncDsl.kt

```diff
@@ -0,0 +1,57 @@
+package io.github.rybalkinsd.kohttp.dsl.async
+
+import io.github.rybalkinsd.kohttp.client.defaultHttpClient
+import io.github.rybalkinsd.kohttp.dsl.context.HttpPutContext
+import io.github.rybalkinsd.kohttp.ext.suspendCall
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.async
+import okhttp3.Call
+import okhttp3.Response
+
+
+/**
+ * Method provides an asynchronous DSL call of HTTP PUT
+ *
+ * @return a deferred `Response` instance.
+ *
+ * Usage example using the default `defaultHttpClient`:
+ *
+ *  <pre>
+ *  val response: Deferred<Response> = httpPutAsync {
+ *      host = "yourhost"
+ *      scheme = "https"
+ *      port = 8080
+ *      path = "path/to/resource"
+ *      param {
+ *          "your param" to "value"
+ *      }
+ *      header { ... }
+ *      body { ... }
+ *  }
+ *  response.await().use { ... }
+ *  </pre>
+ *
+ *  @param client allow to use your own implementation of HttpClient.
+ * `defaultHttpClient` is used by default.
+ *
+ * <pre>
+ *  val response: Deferred<Response> = httpPutAsync(customHttpClient) {
+ *      ...
+ *  }
+ * </pre>
+ *
+ * @see Response
+ * @see HttpPutContext
+ *
+ * @since 0.10.0
+ * @author evgeny
+ */
+fun httpPutAsync(
```

Comment:
any reason to break lines in this way?
what is the default *kotlin* way to make line breaks here?

Reply:
Here what kotlin docs says:
https://kotlinlang.org/docs/reference/coding-conventions.html#function-formatting
```
fun longMethodName(
    argument: ArgumentType = defaultValue,
    argument2: AnotherArgumentType
): ReturnType {
    // body
}
```